S7.Net/PlcAsynchronous.cs

```diff
@@ -54,15 +78,12 @@ public async Task OpenAsync(CancellationToken cancellationToken = default)
             if (s7data.Length < 20)
                 throw new WrongNumberOfBytesException("Not enough data received in response to Communication Setup");
 
-            MaxPDUSize = (short)(s7data[18] * 256 + s7data[19]);
-        }
-
-        private async Task ConnectAsync()
-        {
-            tcpClient = new TcpClient();
-            ConfigureConnection();
-            await tcpClient.ConnectAsync(IP, Port);
-            _stream = tcpClient.GetStream();
+            // TODO: check if this should not rather be UInt16.
```

Comment:
No, it's a byte; see [RFC905](https://tools.ietf.org/html/rfc905), section 13.3.4 b.

Reply:
Ok so this doesn't really make sense to me. We negotiate high PDU sizes > 255 with Snap7 or modern S7 PLCs. How would that be possible if this is only 8bit?

The Signed Part really did not make sense to me / I was sceptical about, since those are quite rarely used in such low level protocols and especially not for a size type.

Of course using signed here doe not matter unless you get a value above 2**15, but it would still be better to just handle it correctly.

Reply:
Also the MaxPDUSize bytes are at address 18 of the PDU Payload and that is usually deep into the S7 protocol, at least for the other functions. So I am not really sure if this matches your reference in the RFC document. But I am really not good at reading this thing.

Reply:
The byte identifies PDU sizes (instead of giving a PDU value), this is from the RFC:
```
0000 1101  8192 octets (not allowed in Class 0)
0000 1100  4096 octets (not allowed in Class 0)
0000 1011  2048 octets
0000 1010  1024 octets
0000 1001   512 octets
0000 1000   256 octets
0000 0111   128 octets
```

Now take the decimal values of above, and perform 2 ^ [decimal value]:
```
binary     size          dec.   2 ^ dec.
0000 1101  8192 octets   13     8192
0000 1100  4096 octets   12     4096
0000 1011  2048 octets   11     2048
0000 1010  1024 octets   10     1024
0000 1001   512 octets    9      512
0000 1000   256 octets    8      256
0000 0111   128 octets    7      128
```

I actually looked into Sally7 to verify this and I must admit I was using the same logic as applied here (UInt 16 from 2 bytes), so honestly I have no clue whether the S7 protocol adheres to the RFC. I guess this is something to investigate.

The Connection Request is actually different from Data Transfer (which I believe is used for read/write), Connection Request mostly or entirely adheres to the RFC while read/write uses a custom protocol inside DT messages.

Reply:
I changed it to just use uint16 big endian parsing for now. This also matches your current Sally7 implementation.

We should probably split further investigating of this into a separate issue. Might require some investigating with different PLC types or analyzing other programs using the S7 protocol.

Reply:
I opened #327 for this.